rosbag2_cpp/src/rosbag2_cpp/writers/sequential_writer.cpp

```diff
@@ -145,16 +145,18 @@ void SequentialWriter::open(
     throw std::runtime_error{error.str()};
   }
 
-  use_cache_ = storage_options.max_cache_size > 0u;
+  use_cache_ = storage_options.max_cache_size > 0u ||
+    (storage_options.snapshot_mode && storage_options.snapshot_duration.nanoseconds() > 0);
```

Comment:
checking `storage_options.snapshot_mode` is redundant?

```suggestion
  use_cache_ = storage_options.max_cache_size > 0u || storage_options.snapshot_duration.nanoseconds() > 0;
```

Reply:
Removed extra check for the `storage_options.snapshot_mode`